### PR TITLE
[doc] update xDS feature matrix

### PR DESCRIPTION
**Pull request by markdroth**
- add entry for aggregate and logical DNS cluster support
- add entry for least_request LB policy
- add a bunch of missing versions in various languages

### CODE REVIEW

**Review by murgatroid99 — APPROVED** [review]

File: doc/grpc_xds_features.md [hunk 1]
```diff
@@ -63,13 +63,15 @@ Support for [xDS v2 APIs](https://www.envoyproxy.io/docs/envoy/latest/api/api_su
...
 [Authorization (RBAC)](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/rbac/v3/rbac.proto):<br><ul><li>`LOG` action has no effect<li>CEL unsupported and rejected</ul> | [A41](https://github.com/grpc/proposal/blob/master/A41-xds-rbac.md) | v1.51.0<br>(C++ and Python) | v1.42.0 | 1.42.0 | |
-[Outlier Detection](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/outlier):<br>Only the following detection types are supported:<ul><li>Success Rate</li><li>Failure Percentage</li></ul> | [A50](https://github.com/grpc/proposal/blob/master/A50-xds-outlier-detection.md) | v1.51.0 | v1.49.0 | v1.50.0 | v1.7.0 | |
+[Least Request LB Policy](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/load_balancers.html#weighted-least-request) | [A48](https://github.com/grpc/proposal/blob/master/A48-xds-least-request-lb-policy.md) |  |  | v1.58.0 |  |
```
Comment (ejona86) [inline doc/grpc_xds_features.md:71]: v1.48.0 for Java. Not 58

**Review by ejona86 — APPROVED** [review]
One important fix, otherwise LGTM

File: doc/grpc_xds_features.md (hunk 1 above)
Comment (dfawley) [reply doc/grpc_xds_features.md:71]: Also, Java, not Go.

**Review by dfawley — APPROVED** [review]
Approve modulo changing least request to be for Java and 1.48.0.

File: doc/grpc_xds_features.md (hunk 1 above)
Comment (markdroth) [reply doc/grpc_xds_features.md]: Doh!  Sorry, accidentally removed the value for Java instead of the one for Go in a previous commit.

Fixed.

**Pull request merged**